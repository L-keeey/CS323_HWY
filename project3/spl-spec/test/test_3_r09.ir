FUNCTION mod :
PARAM v1
PARAM v2
t2 := v1
t6 := v1
t7 := v2
t4 := t6 / t7
t5 := v2
t3 := t4 * t5
t1 := t2 - t3
v3 := t1
t8 := v3
v4 := t8
t9 := v3
RETURN t9
FUNCTION power :
PARAM v5
PARAM v6
t13 := #1
t14 := v6
t11 := t13 + t14
t12 := v6
t10 := t11 - t12
v7 := t10
LABEL label1 :
t15 := v6
t25 := v7
t26 := v7
t23 := t25 - t26
t24 := #90
t21 := t23 + t24
t22 := #89
t19 := t21 - t22
t20 := #1
t17 := t19 + t20
t18 := #2
t16 := t17 - t18
IF t15 > t16 GOTO label2
GOTO label3
LABEL label2 :
t29 := v7
t30 := v5
t28 := t29 * t30
v7 := t28
t27 := v7
t39 := #2
t40 := #1
t37 := t39 * t40
t38 := v6
t35 := t37 * t38
t41 := #1
t42 := v6
t36 := t41 * t42
t33 := t35 - t36
t34 := #1
t32 := t33 - t34
v6 := t32
t31 := v6
GOTO label1
LABEL label3 :
t43 := v7
RETURN t43
FUNCTION getNumDigits :
PARAM v8
t44 := #0
v9 := t44
t45 := v8
t46 := #0
IF t45 < t46 GOTO label4
GOTO label5
LABEL label4 :
t48 := #1
t47 := #0 - t48
RETURN t47
LABEL label5 :
LABEL label6 :
t49 := v8
t50 := #0
IF t49 > t50 GOTO label7
GOTO label8
LABEL label7 :
t53 := v8
t54 := #10
t52 := t53 / t54
v8 := t52
t51 := v8
t57 := v9
t58 := #2
t56 := t57 + t58
v9 := t56
t55 := v9
t61 := v9
t62 := #2
t60 := t61 + t62
v9 := t60
t59 := v9
t65 := v9
t66 := #3
t64 := t65 - t66
v9 := t64
t63 := v9
GOTO label6
LABEL label8 :
t67 := v9
RETURN t67
FUNCTION isNarcissistic :
PARAM v10
t72 := #1
t73 := v10
t70 := t72 + t73
t71 := #1
t69 := t70 - t71
ARG t69
t68 := CALL getNumDigits
v11 := t68
t74 := #0
v12 := t74
t75 := v10
v13 := t75
LABEL label9 :
t76 := v13
t77 := #0
IF t76 > t77 GOTO label10
GOTO label11
LABEL label10 :
ARG #10
ARG v13
t79 := CALL mod
&v14 := t79
t78 := &v14
t84 := v13
t85 := &v14
t82 := t84 - t85
t83 := #10
t81 := t82 / t83
v13 := t81
t80 := v13
t88 := v12
ARG v11
ARG &v14
t89 := CALL power
t87 := t88 + t89
v12 := t87
t86 := v12
GOTO label9
LABEL label11 :
t90 := v12
t91 := v10
IF t90 == t91 GOTO label12
GOTO label13
LABEL label12 :
t92 := #1
RETURN t92
GOTO label14
LABEL label13 :
t93 := #0
RETURN t93
LABEL label14 :
FUNCTION main :
t94 := #0
v15 := t94
t95 := #300
v16 := t95
LABEL label15 :
t96 := v16
t97 := #500
IF t96 < t97 GOTO label16
GOTO label17
LABEL label16 :
ARG v16
t98 := CALL isNarcissistic
t99 := #1
IF t98 == t99 GOTO label18
GOTO label19
LABEL label18 :
t100 := v16
WRITE t100
t103 := v15
t104 := #1
t102 := t103 + t104
v15 := t102
t101 := v15
LABEL label19 :
t107 := v16
t108 := #1
t106 := t107 + t108
v16 := t106
t105 := v16
GOTO label15
LABEL label17 :
t109 := v15
WRITE t109
t110 := v15
RETURN t110
