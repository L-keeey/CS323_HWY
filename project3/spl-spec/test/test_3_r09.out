Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: mod
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: number2
          COMMA
          VarList (1)
            ParamDec (1)
              Specifier (1)
                TYPE: int
              VarDec (1)
                ID: m2
        RP
      CompSt (2)
        LC
        DefList (3)
          Def (3)
            Specifier (3)
              TYPE: int
            DecList (3)
              Dec (3)
                VarDec (3)
                  ID: result
                ASSIGN
                Exp (3)
                  Exp (3)
                    ID: number2
                  MINUS
                  Exp (3)
                    Exp (3)
                      Exp (3)
                        ID: number2
                      DIV
                      Exp (3)
                        ID: m2
                    MUL
                    Exp (3)
                      ID: m2
            SEMI
          DefList (4)
            Def (4)
              Specifier (4)
                TYPE: int
              DecList (4)
                Dec (4)
                  VarDec (4)
                    ID: result2
                  ASSIGN
                  Exp (4)
                    ID: result
              SEMI
        StmtList (5)
          Stmt (5)
            RETURN
            Exp (5)
              ID: result
            SEMI
        RC
    ExtDefList (8)
      ExtDef (8)
        Specifier (8)
          TYPE: int
        FunDec (8)
          ID: power
          LP
          VarList (8)
            ParamDec (8)
              Specifier (8)
                TYPE: int
              VarDec (8)
                ID: base1
            COMMA
            VarList (8)
              ParamDec (8)
                Specifier (8)
                  TYPE: int
                VarDec (8)
                  ID: p1
          RP
        CompSt (8)
          LC
          DefList (9)
            Def (9)
              Specifier (9)
                TYPE: int
              DecList (9)
                Dec (9)
                  VarDec (9)
                    ID: ret1
                  ASSIGN
                  Exp (9)
                    Exp (9)
                      Exp (9)
                        INT: 1
                      PLUS
                      Exp (9)
                        ID: p1
                    MINUS
                    Exp (9)
                      ID: p1
              SEMI
          StmtList (10)
            Stmt (10)
              WHILE
              LP
              Exp (10)
                Exp (10)
                  ID: p1
                GT
                Exp (10)
                  LP
                  Exp (10)
                    Exp (10)
                      Exp (10)
                        Exp (10)
                          Exp (10)
                            Exp (10)
                              ID: ret1
                            MINUS
                            Exp (10)
                              ID: ret1
                          PLUS
                          Exp (10)
                            INT: 90
                        MINUS
                        Exp (10)
                          INT: 89
                      PLUS
                      Exp (10)
                        INT: 1
                    MINUS
                    Exp (10)
                      INT: 2
                  RP
              RP
              Stmt (11)
                CompSt (11)
                  LC
                  StmtList (12)
                    Stmt (12)
                      Exp (12)
                        Exp (12)
                          ID: ret1
                        ASSIGN
                        Exp (12)
                          Exp (12)
                            ID: ret1
                          MUL
                          Exp (12)
                            ID: base1
                      SEMI
                    StmtList (13)
                      Stmt (13)
                        Exp (13)
                          Exp (13)
                            ID: p1
                          ASSIGN
                          Exp (13)
                            Exp (13)
                              Exp (13)
                                Exp (13)
                                  Exp (13)
                                    INT: 2
                                  MUL
                                  Exp (13)
                                    INT: 1
                                MUL
                                Exp (13)
                                  ID: p1
                              MINUS
                              Exp (13)
                                Exp (13)
                                  INT: 1
                                MUL
                                Exp (13)
                                  ID: p1
                            MINUS
                            Exp (13)
                              INT: 1
                        SEMI
                  RC
            StmtList (15)
              Stmt (15)
                RETURN
                Exp (15)
                  ID: ret1
                SEMI
          RC
      ExtDefList (18)
        ExtDef (18)
          Specifier (18)
            TYPE: int
          FunDec (18)
            ID: getNumDigits
            LP
            VarList (18)
              ParamDec (18)
                Specifier (18)
                  TYPE: int
                VarDec (18)
                  ID: number3
            RP
          CompSt (19)
            LC
            DefList (20)
              Def (20)
                Specifier (20)
                  TYPE: int
                DecList (20)
                  Dec (20)
                    VarDec (20)
                      ID: ret3
                    ASSIGN
                    Exp (20)
                      INT: 0
                SEMI
            StmtList (21)
              Stmt (21)
                IF
                LP
                Exp (21)
                  Exp (21)
                    ID: number3
                  LT
                  Exp (21)
                    INT: 0
                RP
                Stmt (22)
                  RETURN
                  Exp (22)
                    MINUS
                    Exp (22)
                      INT: 1
                  SEMI
              StmtList (23)
                Stmt (23)
                  WHILE
                  LP
                  Exp (23)
                    Exp (23)
                      ID: number3
                    GT
                    Exp (23)
                      INT: 0
                  RP
                  Stmt (23)
                    CompSt (23)
                      LC
                      StmtList (24)
                        Stmt (24)
                          Exp (24)
                            Exp (24)
                              ID: number3
                            ASSIGN
                            Exp (24)
                              Exp (24)
                                ID: number3
                              DIV
                              Exp (24)
                                INT: 10
                          SEMI
                        StmtList (25)
                          Stmt (25)
                            Exp (25)
                              Exp (25)
                                ID: ret3
                              ASSIGN
                              Exp (25)
                                Exp (25)
                                  ID: ret3
                                PLUS
                                Exp (25)
                                  INT: 2
                            SEMI
                          StmtList (26)
                            Stmt (26)
                              Exp (26)
                                Exp (26)
                                  ID: ret3
                                ASSIGN
                                Exp (26)
                                  Exp (26)
                                    ID: ret3
                                  PLUS
                                  Exp (26)
                                    INT: 2
                              SEMI
                            StmtList (27)
                              Stmt (27)
                                Exp (27)
                                  Exp (27)
                                    ID: ret3
                                  ASSIGN
                                  Exp (27)
                                    Exp (27)
                                      ID: ret3
                                    MINUS
                                    Exp (27)
                                      INT: 3
                                SEMI
                      RC
                StmtList (29)
                  Stmt (29)
                    RETURN
                    Exp (29)
                      ID: ret3
                    SEMI
            RC
        ExtDefList (32)
          ExtDef (32)
            Specifier (32)
              TYPE: int
            FunDec (32)
              ID: isNarcissistic
              LP
              VarList (32)
                ParamDec (32)
                  Specifier (32)
                    TYPE: int
                  VarDec (32)
                    ID: number4
              RP
            CompSt (33)
              LC
              DefList (34)
                Def (34)
                  Specifier (34)
                    TYPE: int
                  DecList (34)
                    Dec (34)
                      VarDec (34)
                        ID: numDigits4
                      ASSIGN
                      Exp (34)
                        ID: getNumDigits
                        LP
                        Args (34)
                          Exp (34)
                            Exp (34)
                              Exp (34)
                                INT: 1
                              PLUS
                              Exp (34)
                                ID: number4
                            MINUS
                            Exp (34)
                              INT: 1
                        RP
                  SEMI
                DefList (35)
                  Def (35)
                    Specifier (35)
                      TYPE: int
                    DecList (35)
                      Dec (35)
                        VarDec (35)
                          ID: sum4
                        ASSIGN
                        Exp (35)
                          INT: 0
                    SEMI
                  DefList (36)
                    Def (36)
                      Specifier (36)
                        TYPE: int
                      DecList (36)
                        Dec (36)
                          VarDec (36)
                            ID: n4
                          ASSIGN
                          Exp (36)
                            ID: number4
                      SEMI
                    DefList (37)
                      Def (37)
                        Specifier (37)
                          TYPE: int
                        DecList (37)
                          Dec (37)
                            VarDec (37)
                              ID: s4
                        SEMI
              StmtList (38)
                Stmt (38)
                  WHILE
                  LP
                  Exp (38)
                    Exp (38)
                      ID: n4
                    GT
                    Exp (38)
                      INT: 0
                  RP
                  Stmt (38)
                    CompSt (38)
                      LC
                      StmtList (39)
                        Stmt (39)
                          Exp (39)
                            Exp (39)
                              ID: s4
                            ASSIGN
                            Exp (39)
                              ID: mod
                              LP
                              Args (39)
                                Exp (39)
                                  ID: n4
                                COMMA
                                Args (39)
                                  Exp (39)
                                    INT: 10
                              RP
                          SEMI
                        StmtList (40)
                          Stmt (40)
                            Exp (40)
                              Exp (40)
                                ID: n4
                              ASSIGN
                              Exp (40)
                                Exp (40)
                                  LP
                                  Exp (40)
                                    Exp (40)
                                      ID: n4
                                    MINUS
                                    Exp (40)
                                      ID: s4
                                  RP
                                DIV
                                Exp (40)
                                  INT: 10
                            SEMI
                          StmtList (41)
                            Stmt (41)
                              Exp (41)
                                Exp (41)
                                  ID: sum4
                                ASSIGN
                                Exp (41)
                                  Exp (41)
                                    ID: sum4
                                  PLUS
                                  Exp (41)
                                    ID: power
                                    LP
                                    Args (41)
                                      Exp (41)
                                        ID: s4
                                      COMMA
                                      Args (41)
                                        Exp (41)
                                          ID: numDigits4
                                    RP
                              SEMI
                      RC
                StmtList (44)
                  Stmt (44)
                    IF
                    LP
                    Exp (44)
                      Exp (44)
                        ID: sum4
                      EQ
                      Exp (44)
                        ID: number4
                    RP
                    Stmt (45)
                      RETURN
                      Exp (45)
                        INT: 1
                      SEMI
                    ELSE
                    Stmt (47)
                      RETURN
                      Exp (47)
                        INT: 0
                      SEMI
              RC
          ExtDefList (50)
            ExtDef (50)
              Specifier (50)
                TYPE: int
              FunDec (50)
                ID: main
                LP
                RP
              CompSt (50)
                LC
                DefList (51)
                  Def (51)
                    Specifier (51)
                      TYPE: int
                    DecList (51)
                      Dec (51)
                        VarDec (51)
                          ID: count
                        ASSIGN
                        Exp (51)
                          INT: 0
                    SEMI
                  DefList (52)
                    Def (52)
                      Specifier (52)
                        TYPE: int
                      DecList (52)
                        Dec (52)
                          VarDec (52)
                            ID: i
                          ASSIGN
                          Exp (52)
                            INT: 300
                      SEMI
                StmtList (53)
                  Stmt (53)
                    WHILE
                    LP
                    Exp (53)
                      Exp (53)
                        ID: i
                      LT
                      Exp (53)
                        INT: 500
                    RP
                    Stmt (54)
                      CompSt (54)
                        LC
                        StmtList (55)
                          Stmt (55)
                            IF
                            LP
                            Exp (55)
                              Exp (55)
                                ID: isNarcissistic
                                LP
                                Args (55)
                                  Exp (55)
                                    ID: i
                                RP
                              EQ
                              Exp (55)
                                INT: 1
                            RP
                            Stmt (56)
                              CompSt (56)
                                LC
                                StmtList (57)
                                  Stmt (57)
                                    Exp (57)
                                      ID: write
                                      LP
                                      Args (57)
                                        Exp (57)
                                          ID: i
                                      RP
                                    SEMI
                                  StmtList (58)
                                    Stmt (58)
                                      Exp (58)
                                        Exp (58)
                                          ID: count
                                        ASSIGN
                                        Exp (58)
                                          Exp (58)
                                            ID: count
                                          PLUS
                                          Exp (58)
                                            INT: 1
                                      SEMI
                                RC
                          StmtList (60)
                            Stmt (60)
                              Exp (60)
                                Exp (60)
                                  ID: i
                                ASSIGN
                                Exp (60)
                                  Exp (60)
                                    ID: i
                                  PLUS
                                  Exp (60)
                                    INT: 1
                              SEMI
                        RC
                  StmtList (62)
                    Stmt (62)
                      Exp (62)
                        ID: write
                        LP
                        Args (62)
                          Exp (62)
                            ID: count
                        RP
                      SEMI
                    StmtList (63)
                      Stmt (63)
                        RETURN
                        Exp (63)
                          ID: count
                        SEMI
                RC
Program
ExtDefList
ExtDef
FunDec
FUNCTION mod :
VarList
ParamDec
VarDec
PARAM v1
VarList
ParamDec
VarDec
PARAM v2
ComSt
DefList
Def
DecList
Dec
Exp
Exp
t2 := v1
Exp
Exp
Exp
t6 := v1
Exp
t7 := v2
t4 := t6 / t7
Exp
t5 := v2
t3 := t4 * t5
t1 := t2 - t3
v3 := t1
DefList
Def
DecList
Dec
Exp
